pyfakefs/fake_filesystem.py

```diff
@@ -1245,6 +1250,49 @@ def CreateLink(self, file_path, link_target):
     return self.CreateFile(resolved_file_path, st_mode=stat.S_IFLNK | PERM_DEF,
                            contents=link_target)
 
+  def CreateHardLink(self, old_path, new_path):
+    """Create a hard link at new_path, pointing at old_path.
+
+    Args:
+      old_path: an existing link to the target file
+      new_path: the destination path to create a new link at
+
+    Returns:
+      the FakeFile object referred to by old_path
+
+    Raises:
+      OSError:  if something already exists at new_path
+      OSError:  if the parent directory doesn't exist
+    """
+    new_path_normalized = self.NormalizePath(new_path)
+    if self.Exists(new_path_normalized):
+      raise IOError(errno.EEXIST,
```

Comment:
Yeah, fair enough. I checked what `ln` reports, for inspiration, and they make the same mistake. I think "fake filesystem object" might be confusing for some peope. How about `Fake file or directory already exists at path`?